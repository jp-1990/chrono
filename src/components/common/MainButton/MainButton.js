import React from "react";
import { Pressable, Text } from "react-native";

const MainButton = ({ width, colorBG, colorText, label, ripple, onPress }) => {
  return (
    <Pressable
      style={{
        backgroundColor: colorBG,
        width: width,
        justifyContent: "center",
        alignItems: "center",
        height: 42,
        marginTop: 36,
      }}
      android_ripple={{
        color: ripple,
        borderless: false,
        radius: 150,
      }}
      onPress={onPress ? () => onPress() : null}
    >
      <Text
        style={{
          color: colorText,
          fontFamily: "lato-regular",
          fontSize: 18,
        }}
      >
        {label}
      </Text>
    </Pressable>
  );
};

export default MainButton;
