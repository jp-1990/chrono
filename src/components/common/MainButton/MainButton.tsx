import React from "react";
import { Pressable, Text } from "react-native";

interface Props {
  onPress(): void;
  label: string;
  width: number | string;
  ripple: string;
  colorBG: string;
  colorText: string;
  marginTop?: number;
}

const MainButton: React.FC<Props> = ({
  width,
  colorBG,
  colorText,
  label,
  ripple,
  onPress,
  marginTop,
}) => {
  return (
    <Pressable
      style={{
        backgroundColor: colorBG,
        width: width,
        justifyContent: "center",
        alignItems: "center",
        height: 42,
        marginTop: marginTop || 36,
      }}
      android_ripple={{
        color: ripple,
        borderless: false,
        radius: 150,
      }}
      onPress={onPress ? () => onPress() : null}
    >
      <Text
        style={{
          color: colorText,
          fontFamily: "lato-regular",
          fontSize: 18,
        }}
      >
        {label}
      </Text>
    </Pressable>
  );
};

export default MainButton;
