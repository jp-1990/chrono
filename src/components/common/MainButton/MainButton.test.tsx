import React from "react";
import TestRenderer from "react-test-renderer";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

import MainButton from "./MainButton";

describe("<MainButton />", () => {
  afterEach(cleanup);

  const mockCallback = jest.fn();
  const testProps = { label: "testing", onPress: mockCallback };
  const testRender = TestRenderer.create(<MainButton {...testProps} />);

  it("renders correctly", () => {
    expect(testRender).toMatchSnapshot();
  });

  it("render contains label prop", () => {
    const { getByText } = render(<MainButton {...testProps} />);
    expect(getByText("testing")).toBeTruthy();
  });

  it("uses the onPress prop when it is pressed", () => {
    const { getByText } = render(<MainButton {...testProps} />);
    fireEvent.press(getByText("testing"));
    expect(mockCallback.mock.calls.length).toEqual(1);
    mockCallback.mockClear();
  });

  it("does not use the onPress prop when it is not pressed", () => {
    const { getByText } = render(<MainButton {...testProps} />);
    expect(getByText("testing")).toBeTruthy();
    expect(mockCallback.mock.calls.length).toEqual(0);
    mockCallback.mockClear();
  });
});
