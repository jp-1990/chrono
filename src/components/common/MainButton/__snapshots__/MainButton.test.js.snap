// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "height": 42,
      "justifyContent": "center",
      "marginTop": 36,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": undefined,
        "fontFamily": "lato-regular",
        "fontSize": 18,
      }
    }
  >
    testing
  </Text>
</View>
`;
